test/distributed/_tensor/test_utils.py

```diff
@@ -122,6 +123,17 @@ def test_compute_local_shape_and_global_offset_2D(self):
                     global_tensor[dim0_start:dim0_end, dim1_start:dim1_end],
                 )
 
+    @with_comms
+    def test_compute_global_padding(self):
+        local_tensor = torch.randn(8, requires_grad=True)
+        mesh = init_device_mesh(self.device_type, (3, 2))
+        dist_tensor = distribute_tensor(local_tensor, mesh, [Shard(0), Shard(0)])
+        padding = compute_global_padding(
+            dist_tensor.shape, dist_tensor.device_mesh, dist_tensor.placements
+        )
+        self.assertEqual(len(padding), dist_tensor.ndim)
+        self.assertEqual(padding, [4])
```

Comment:
n00b question. Why should it be 4? My understanding of this test is:
shard a tensor of size [8] into 6 ranks (device mesh: `[[0, 1], [2, 3], [4, 5]]`)by `chunk()`, the first 4 shards are of size [2] and the other 2 shards are empty. Shouldn't the padding be [0] on rank 0, 1, 2, 3; and [2] on rank 4 and 5?

BTW, what is the expected result of sharding size [9] tensor into this 3x2 mesh? The padded size should still be 2 on all ranks. Which ranks will have padding? My question is mainly about the order of putting first unpadded tensors when sharding one tensor dim over multiple mesh dims.

Reply:
Good question. I added some summary in the PR to show what we are trying to do. Basically, we want to explore doing padding before sharding. See if the summary gives a better idea and we can discuss this offline as well.